applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_hydraulic_solver.py

```diff
@@ -0,0 +1,692 @@
+# Importing the Kratos Library
+import KratosMultiphysics
+import KratosMultiphysics.kratos_utilities as KratosUtilities
+
+# Import applications
+import KratosMultiphysics.FluidDynamicsApplication as KratosCFD
+
+# Import base class file
+from KratosMultiphysics.FluidDynamicsApplication.fluid_solver import FluidSolver
+
+
+from pathlib import Path
+
+def CreateSolver(model, custom_settings):
+    return NavierStokesTwoFluidsHydraulicSolver(model, custom_settings)
+
+class NavierStokesTwoFluidsHydraulicSolver(FluidSolver):
+
+    @classmethod
+    def GetDefaultParameters(cls):
+        ##settings string in json format
+        default_settings = KratosMultiphysics.Parameters("""
+        {
+            "solver_type": "two_fluids_hydraulic",
+            "model_part_name": "",
+            "domain_size": -1,
+            "model_import_settings": {
+                "input_type": "mdpa",
+                "input_filename": "unknown_name",
+                "reorder": false
+            },
+            "material_import_settings": {
+                "materials_filename": ""
+            },
+            "distance_reading_settings"    : {
+                "import_mode"         : "from_mdpa",
+                "distance_file_name"  : "no_distance_file"
+            },
+            "maximum_iterations": 7,
+            "echo_level": 0,
+            "time_order": 2,
+            "time_scheme": "NEEDS TO BE FIXED --> PROBLEMS CONVERGENCE CRITERIO",
+            "compute_reactions": false,
+            "analysis_type": "non_linear",
+            "reform_dofs_at_each_step": false,
+            "consider_periodic_conditions": false,
+            "relative_velocity_tolerance": 1e-3,
+            "absolute_velocity_tolerance": 1e-5,
+            "relative_pressure_tolerance": 1e-3,
+            "absolute_pressure_tolerance": 1e-5,
+            "linear_solver_settings"       : {
+                "solver_type"         : "amgcl"
+            },
+            "volume_model_part_name" : "volume_model_part",
+            "skin_parts": [""],
+            "assign_neighbour_elements_to_conditions": true,
+            "no_skin_parts":[""],
+            "time_stepping"                : {
+                "automatic_time_step" : true,
+                "CFL_number"          : 1,
+                "minimum_delta_time"  : 1e-2,
+                "maximum_delta_time"  : 1.0,
+                "time_step"           : 0.0
+            },
+            "periodic": "periodic",
```

Comment:
This is the option to turn on/off the artificicial viscosity in order to avoid numerical noise. In addition there is a limiter in case we want to limit the maximum amount of artificial viscosity